en/concept/custom_mixer_payload.md

```diff
@@ -0,0 +1,77 @@
+# Custom Payload Mixers
+
+This topic shows how to add a custom [mixer](../concept/mixing.md) for programmatically controlling a custom payload (e.g., an electromagnetic gripper).
+
+The topic is intended for developers who want to support payload types that do not have existing control group definitions (e.g. gimbals have a control group, but grippers do not).
+You should already have read [Mixing and Actuators](../concept/mixing.md).
+
+
+## Payload Mixer Example
+
+A payload mixer is just a [simple mixer](../concept/mixing.md#simple-mixer) that maps any of the function values from [Control Group #6 (First Payload)](../concept/mixing.md#control_group_6) to a particular output.
+You can then publish uORB topics to the selected control group function and their value will be mapped to the specified output.
+
+For this example, we'll create a custom mixer based on the *RC passthrough mixer* ([pass.aux.mix](https://github.com/PX4/Firmware/blob/master/ROMFS/px4fmu_common/mixers/pass.aux.mix)).
+This mixer is commonly loaded into the AUX PWM ports on large multicopters). 
+It passes through the values of 4 user-defined RC channels (set using the [RC_MAP_AUXx/RC_MAP_FLAPS](../advanced/parameter_reference.md#RC_MAP_AUX1) parameters) to the first four outputs on the AUX PWM output.
+
+```
+# Manual pass through mixer for servo outputs 1-4
+
+# AUX1 channel (select RC channel with RC_MAP_AUX1 param)
+M: 1
+S: 3 5  10000  10000      0 -10000  10000
+
+# AUX2 channel (select RC channel with RC_MAP_AUX2 param)
+M: 1
+S: 3 6  10000  10000      0 -10000  10000
+
+# AUX3 channel (select RC channel with RC_MAP_AUX3 param)
+M: 1
+S: 3 7  10000  10000      0 -10000  10000
+
+# FLAPS channel (select RC channel with RC_MAP_FLAPS param)
+M: 1
+S: 3 4  10000  10000      0 -10000  10000
+```
+
+> **Note** The file defines a [simple mixer](../concept/mixing.md#simple-mixer) with four outputs.
+> - `M: 1` indicates an output that is defined by one control input (the following `S` line).
+> - `S: 3 `_`n`_ indicates that the input is the n<>th<> input of [Control Group 3 (Manual Passthrough)](../concept/mixing.md#control-group-3-manual-passthrough).
+>  So for `S: 3 5` the input is called "RC aux1", and will output the value of the channel mapped by parameter `RC_MAP_AUX1`.
+> - The order of output declaration of 
+
+
+Start by copying the mixer file and putting it onto the SD Card at: **/etc/mixers/pass.aux.mix** (see [Mixing and Actuators > Loading a Custom Mixer](../concept/mixing.md#loading_custom_mixer).
```

Comment:
Worth noting that `pass.aux.mix` depends on the airframe. `pass` is typically the default, but in particular VTOL's typically have their own.
This can be checked in the airframe config: https://github.com/PX4/Firmware/blob/master/ROMFS/px4fmu_common/init.d/airframes/13006_vtol_standard_delta#L58.
I now see you explained this further down though.

Reply:
Actually I didn't completely explain this because it isn't clear what the default is in all cases. Based on comments and what I see in the airfames reference I think it looks like this:

Is this right
1. Set MAIN mixer : `set MIXER xxxx`.
1. Explicitly set AUX MIXER: `set MIXER_AUX yyyyy`
1. If no MIXER_AUX explicitly set:
   - **FOR copter, FW, etc. (not VTOL)**, then MIXER name is used as prefix for expected aux file.
  - **FOR VTOL airframe** then `pass.aux.mix` used for AUX.

Is that right?


    
  ```
This implies that to set the main mixer we call `set MIXER` to set AUX

Reply:
Almost, for AUX it is:
- fw+mc use pass by default
- vtol uses none by default
- if MIXER_AUX is none (so vtol+airframe does not set it), then `set MIXER` name is used as prefix for expected AUX file.
- there's also some logic for gimbals: https://github.com/PX4/Firmware/blob/master/ROMFS/px4fmu_common/init.d/rc.interface#L25

Reply:
Awesome, thanks. I've re-written this section to split out mixer loading and custom mixer loading. I think very clear now what gets loaded when. Please recheck.